### ISSUE TITLE
gdalwarp - NoDataVal Erroneous Outputs

### ISSUE THREAD

**Issue by bryantcurto** [OSGeo/gdal#1724]
Hello! 

I've been trying to use `gdalbuildvrt` and `gdalwarp` to merge, project, and crop a set of overlapping images with a new resolution.

## Expected behavior and actual behavior.
I've been running into an issue where the edges of the resulting image have nondeterministic, erroneous values (ie. very large and small values) and values in regions that I would expect only NoDataValues. The images are Float32, 1 banded, and have nodata = -340282346638528859811704183484516925440.

## Steps to reproduce the problem.
```
gdalbuildvrt -r cubic -srcnodata -340282346638528859811704183484516925440 -vrtnodata -340282346638528859811704183484516925440 -te -122.445455 37.747147 -122.433587 37.756551 -tr 0.000031 0.000031 -input_file_list file-list.txt raster.vrt 
gdalwarp --config GDAL_CACHEMAX 1024 -t_srs "+proj=utm +datum=WGS84 +zone=10 +units=m" -tr 2.709812 3.424375 -te 548854.456592 4177905.79342 549893.700272 4178955.41704 -r cubic -srcnodata -340282346638528859811704183484516925440 -dstnodata 0 raster.vrt raster.tif 
```
Here is a [zip of input files (file-list.txt, raster1.tif, raster2.tif, raster3.tif)](https://github.com/OSGeo/gdal/files/3397816/input-files.zip) for the above commands.

In the region bounded by my extents, two images (raster1.tif and raster3.tif) have only NoDataValues, while the other has valid values for part of the region.

Some other things I've discovered while attempting to debug this issue: 
- Setting the resampling method of `gdalbuildvrt` to anything besides nearest results in the issue. 
- Initializing the destination image in gdalwarp via `-wo "INIT_DEST=NO_DATA"` fails to resolve this issue.

## Operating system

Ubuntu 16.04 64 bit

## GDAL version and provenance

In my testing, I've found that this issue does not arise when using gdal v2.2.0 or v2.2.4, but does with v2.3.0 and v2.4.0.

Any help would be appreciated!
[Here](http://osgeo-org.1560.x6.nabble.com/gdal-dev-gdalwarp-Erroneous-Outputs-td5401208.html) is my post on the same issue on the GDAL forum.

**Comment by rouault** [comment]
That should work with GDAL 2.4.2 without speciyfing -srcnodata / -vrtnodata arguments at all.
With older versions, you may need to edit the .vrt file to have exactly <NoDataValue>-3.4028234663852886e+38</NoDataValue> and <NODATA>-3.4028234663852886e+38</NODATA>

**Comment by bryantcurto** [comment]
@rouault, thanks for the quick response!

I tested the MWE I linked on both GDAL v2.4.2 and commit b329312 (which references this Issue), both with and without specifying the -srcnodata / -vrtnodata arguments. All tests produced erroneous values (including NaN).

**Comment by bryantcurto** [comment]
@rouault, bump

**Comment by bryantcurto** [comment]
This issue appears to be resolved in GDAL 2.4.3.
